src/containers/Home/index.js

```diff
@@ -1,78 +1,155 @@
 import { useEffect, useState } from "react";
-import { getAllProducts } from "../../utils/databaseQuery";
+import { ProductProvider, useProducts } from "./productContext";
 
 import "./index.css";
 
 const Home = () => {
-  const [products, setProducts] = useState([]);
+  /* Custom webhook for products */
+  const { productFilters, dispatch } = useProducts();
+
+  /* States for product filters */
+  const [genderFilter, setGenderFilter] = useState([]);
+  const [starFilter, setStarFilter] = useState([]);
+  const [minPrice, setMinPrice] = useState(1);
+
+  /* Function to handle various filter */
+  const handleFilter = (type) => {
+    if (type === "gender") {
+      return function (e) {
+        if (e.target.checked)
+          setGenderFilter([...genderFilter, e.target.value]);
+        else
+          setGenderFilter([
+            ...genderFilter.filter((item) => item !== e.target.value),
+          ]);
+      };
+    } else if (type === "rating") {
+      return function (e) {
+        if (e.target.checked)
+          setStarFilter([...new Set([...starFilter, +e.target.value[0]])]);
```

Comment:
nice , what is this, what does this do. DM me.

Reply:
Done, shared the info